ammber/BinarySystems.py

```diff
@@ -476,3 +479,307 @@ def get_lower_convex_hull(inputpoints):
+        if xdist_dict is None:
+            xdist_dict = {T: 0.001 for T in self.isothermal_systems.keys()}
+        if npts_dict is None:
+            npts_dict = {T: 101 for T in self.isothermal_systems.keys()}
+
+        # Iterate through isothermal systems and initialize near equilibrium
+        for T, system in self.isothermal_systems.items():
+            if T in kwellmax_dict and T in xdist_dict and T in npts_dict:
+                neareq_system = discrete_systems.isothermal_systems[T].resample_near_equilibrium(x, xdist=xdist_dict[T], npts=npts_dict[T])
+                system.from_discrete(neareq_system, kwellmax=kwellmax_dict[T])
+
+    def to_discrete(self, xdata=None, xrange=(1e-14, 1.0 - 1e-14), npts_dict=None):
+        """
+        Builds discrete systems of phases by evaluating free energies on a range
+        Provide either xdata, or a range and number of points.
+
+        Parameters
+        ----------
+        xdata : list (float)
+            compositions to be sampled
+        xrange : tuple (min, max)
+            range of compositions to be linearly sampled
+        npts : dict
+            number of points to be sampled from in xrange
+
+        Returns
+        -------
+        BinaryDiscretePhase object
+        """
+        discrete_systems = BinaryDiscreteSystem()
+
+        if npts_dict is None:
+            npts_dict = {T: 1001 for T in self.isothermal_systems.keys()}
+        
+        # Add temperatures and corresponding systems to the discrete system
+        for T, system in self.isothermal_systems.items():
+            if T in npts_dict:
+                discrete_systems.add_temperature(T, system.to_discrete(xdata=xdata, xrange=xrange, npts=npts_dict[T]))
+
+        return discrete_systems
+    
+    def fit_parameters(self, phase_name):
+        """
+        Fit the parameters (A, B, D) of a phase.
+
+        Parameters
+        ----------
+        phase_name : str
+            The name of the phase to fit the parameters for.
+        """
+        # Extract kwell, cmin, fmin values for the given phase
+        kwell_list = []
+        cmin_list = []
+        fmin_list = []
+        
+        # Iterate through isothermal systems
+        for temperature in self.isothermal_systems.keys():
+            system = self.isothermal_systems[temperature]
+
+            # Check if the target phase name is contained in any of the keys
+            for ph_name in system.phases.keys():
+                if phase_name in ph_name:
+                    phase_system = system.phases[ph_name]
+                    kwell_list.append((temperature, phase_system.kwell))
+                    cmin_list.append((temperature, phase_system.cmin))
+                    fmin_list.append((temperature, phase_system.fmin))
+                    break  # Found the phase, no need to check further keys
+
+        # Convert lists to dictionaries for kwell, cmin, and fmin
+        kwell_dict = dict(kwell_list)
```

Comment:
These should be attributes of the object.  Right now they aren't stored or returned.